ortho_seq_code/plotclass.py

```diff
@@ -0,0 +1,361 @@
+import os
+import numpy as np
+from ortho_seq_code.constants_orthoseqs import *
+import matplotlib.pyplot as plt
+import click
+from ortho_seq_code.utils import create_dir_if_not_exists
+
+
+class rf1d:
+    # Initialize rf1d object
+    def __init__(
+        self, ndarray, alphbt_input, molecule="protein", phenotype=None, out_dir=None
+    ):
+        try:
+            self.x = ndarray
+            self.x_flat = list(ndarray.flatten())
+            self.s = ndarray.shape[0]
+            self.d = ndarray.shape[1]
+            self.ind = np.arange(self.s)
+            self.num_dm = np.arange(self.d)
+            self.alphbt_input = alphbt_input.split(",")[0 : ndarray.shape[1]]
+            self.m = molecule
+            self.complist = ["<", ">", "<>", "><"]
+            self.phenotype = phenotype
+            self.out_dir = out_dir
+        except:
+            print(
+                "Error: Please provide a valid ndarray object and molecule type when initializing."
+            )
+
+    def summary(self):
+        print("rf1d Object:\n")
```

Comment:
so summary is just printed? could it be written to a file as well?

Reply:
if someone closed their command line before copying it then it would be lost, so it would be ideal if you could write it to a file too